infra/Pos/Communication/Relay/Types.hs

```diff
@@ -44,10 +45,17 @@ instance Buildable PropagationMsg where
     build (DataOnlyPM conts) =
         Buildable.build conts
 
--- | Queue of InvMsges which should be propagated.
-type RelayPropagationQueue = TBQueue (Microsecond, PropagationMsg)
+data RelayContext m = RelayContext
```

Comment:
> This issue is that, since it takes a callback, this m must be the same m that we use for the entire application, like RealMode ssc.

Which callback are you talking about? Do you mean that we should pass `RelayerPolicy` to `rsEnqueue` and `rsDequeue`? I don't see any place where `IO` is in negative position in `RelayerPolicy`.

If the problem is about passing `RelayerPolicy` to `rsEnqueue`, then again I don't think it's a problem. There should be a way to create `RelayerPolicy` (obviously). There might be `createRelayerPolicy :: SomeData -> RelayerPolicy` or `defRelayerPolicy :: RelayerPolicy`. Doesn't matter. In any case if we have some data (maybe empty) in reader context, we can create `RelayerPolicy`. And then we can pass `RelayerPolicy` to `rsEnqueue` or any other function which needs it.

Or did you mean something else by _callback_?